clients/drcachesim/tools/opcode_mix.cpp

```diff
@@ -190,6 +197,21 @@ opcode_mix_t::parallel_shard_memref(void *shard_data, const memref_t &memref)
         memref.data.type != TRACE_TYPE_INSTR_NO_FETCH) {
         return true;
     }
+
+    /* At this point we start processing memref instructions, so we're past the header of
+     * the trace, which contains the trace filetype. If we didn't encounter a
+     * TRACE_MARKER_TYPE_FILETYPE we return an error and stop processing the trace.
+     * XXX i#6812: we could allow traces that have some shards with no filetype, as long
```

Comment:
I agree with updating opcode_mix to use the stream APIs (i#6812). But I don't know whether we want to tolerate "traces that have some shards with no filetype"

Reply:
Clarified that we should spend sometime deciding if we want to allow it first. For now we don't.